R/read.R

```diff
@@ -289,16 +298,57 @@ readH5AD <- function(file, X_name = NULL, use_hdf5 = FALSE,
     mat
 }
 
+.read_convert <- function(file, path, recursive=FALSE) {
+    .convert_special_types(rhdf5::h5read(file, path),
+                           recursive=recursive)
+}
+
+.convert_special_types <- function(obj, recursive=FALSE) {
+    if (identical(names(obj), c('categories', 'codes'))) {
```

Comment:
I would strongly suggest using the attributes specifying the encoding to figure out how to read types rather than inferring from the contents. But maybe this is out of scope for this PR.

https://anndata.readthedocs.io/en/latest/fileformat-prose.html#elements

Reply:
I think it's doable by passing the path here and using `rhdf5::h5_readAttributes()` -- will look into it

Reply:
Renamed to `.convert_element`, added some comments, and used `rhdf5::h5readAttributes` to read the `encoding-type` attribute (passing in the path as well).